docs/specs/alcproxies.md

```diff
@@ -0,0 +1,251 @@
+    }
+}
+```
+
+...and then within the main application...
+
+
+```C#
+...
+
+AppDomainSetup ads = new AppDomainSetup();
+ads.ApplicationBase = AppDomain.CurrentDomain.BaseDirectory;
+
+...
+
+AppDomain ad2 = AppDomain.CreateDomain("AD #2", null, ads);
+
+SampleType mbrt = 
+            (SampleType) ad2.CreateInstanceAndUnwrap(
+                exeAssembly, 
+                typeof(SampleType).FullName
+            );
+			
+mbrt.SomeMethod();
+
+...
+```
+
+This allows for programs and classes to be loaded with an extra boundary of isolation, to help prevent things like unstable code bringing down a process. This was removed due to the support for the implementation (`TransparentProxy`) being fairly expensive.
+
+### AssemblyLoadContext
+Currently, there is no easy way to load an object from an ALC and call methods normally on it. However, you can use reflection like so:
+```C#
+AssemblyLoadContext alc = new AssemblyLoadContext(assemblyPath, true); //Create the ALC
+Assembly aa = alc.LoadFromAssemblyPath(assemblyPath); //Load the assembly into the ALC
+Type classType = aa.GetType("AssemblyName.DesiredType");
+
+// For instantiating below, you can also use Activator.GetInstance(classType) if you know there is a default constructor that takes no arguments
+object obj = (classType.GetConstructor(Type.EmptyTypes).Invoke(new object[] { }));
+
+obj.GetType().GetMethod("SomeMethod").Invoke(obj, null);
+```
+The problem here is that due to Type Isolation of ALCs, you cannot directly cast the instantiated object to its original type or an inherited type, as the CLR believes them to be different types from what's loaded from the ALC. 
+
+There are two ways to use the instance from the ALC:
+* Use reflection to invoke calls of the objects type, gotten directly using object.GetType()
+* Use a shared assembly between the user and target assembly that holds a class/interface that the type you want inherits from, and using said shared type instead of the type you want directly (See [this example](https://github.com/dotnet/samples/tree/master/core/extensions/AppWithPlugin ) to see it working).
+* Use `DispatchProxy` to create a proxy within the user ALC, pointing to an object in the target ALC. This also requires a shared assembly.
+
+Anything requiring a shared assembly is inadequate for what we're trying to do, as one of the main benefits of AppDomains having totally separate types was with version compatibility. In .NET Framework, different versions of assemblies could be loaded into separate AppDomains, and communication could occur between objects as if they were one shared type. For example, if a user has a program that uses dependency A, and wants to load a plugin that has classes that inherit from older versions of dependency A, AppDomains would still be able to send and receive information between the plugin and the main program. If we enforce shared assemblies with this API, we lose the ability to allow for things like this.
```

Comment:
It may also be worth noting that with AppDomains the assembly containing the created type must be loaded into both appdomains, but it allows for other assemblies to be loaded only into the target appdomain. A solution for ALCs that doesn't use shared types would have similar behavior.